build.sh

```diff
@@ -150,24 +156,35 @@ main() {
   if [ -z "$SKIP_PREPARATION" ]; then
     _prepare_foxhound
   fi
-  if [ -n "$WITH_PLAYWRIGHT_INTEGRATION" ] && [ -z "$SKIP_PREPARATION" ]; then
+  if [ -n "$WITH_PLAYWRIGHT_INTEGRATION" ] && [ -z "$SKIP_PATCHING" ]; then
```

Comment:
Hmm, is there a reason why we change it to skip patching instead of skip preparation? I think it would be beneficial to keep the skip preparation logic intact (i.e., if SKIP_PATCHING or SKIP_PREPARATION is set, don't do the playwright preparation logic)

Reply:
What I would like to do is have an option to run the patching part only but *not* the preparation (or any other part of the build process.

An alternative would be keep SKIP_PREPARATION as it is now, and have a FORCE_PATCHING which will patch even if the preparation is skipped. That should keep the script command line options backward compatible but still support my pipeline use case.

How does that sound?

Reply:
What I would like to do is have a way to skip *just* the checkout part of the build script (ie checkout_playwright and prepare_foxhound), as these are handled by the Jenkins script separately:
```
_checkout_playwright                # Handled by Jenkins Script
_prepare_foxhound                   # Handled by Jenkins Script
_check_foxhound_repo_state          # Patch
_prepare_playwright                 # Patch
_patch_foxhound                     # Patch
_determine_obj_dir
_build_foxhound                     # Build
_package_foxhound                   # Package
```
At the moment (ie without my changes), SKIP_PREPARATION skips both the checkout phases *and* the patching. What I would like to have is a way to skip the checkout but then *just* do the patching (ie check_foxhound_repo_state, prepare_playwright  and patch_foxhound).

The idea behind an additional SKIP_PATCHING flag was to allow skipping checkout and patching parts of the script separately.